recipes/hwloc/all/conanfile.py

```diff
@@ -0,0 +1,112 @@
+    url = "https://github.com/conan-io/conan-center-index"
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_libxml2": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_libxml2": False
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        if self.options.with_libxml2:
+            self.requires("libxml2/2.9.12")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def layout(self):
+        if self.settings.os == "Windows":
+            cmake_layout(self, src_folder="src")
+        else:
+            basic_layout(self, src_folder="src")
+
+    def generate(self):
+        if self.settings.os == "Windows":
+            deps = CMakeDeps(self)
+            deps.generate()
+            tc = CMakeToolchain(self)
+            tc.cache_variables["HWLOC_ENABLE_TESTING"] = 'OFF'
+            tc.cache_variables["HWLOC_SKIP_LSTOPO"] = 'ON'
+            tc.cache_variables["HWLOC_SKIP_TOOLS"] = 'ON'
+            tc.cache_variables["HWLOC_SKIP_INCLUDES"] = 'OFF'
+            tc.cache_variables["HWLOC_WITH_OPENCL"] = 'OFF'
+            tc.cache_variables["HWLOC_WITH_CUDA"] = 'OFF'
+            tc.cache_variables["HWLOC_BUILD_SHARED_LIBS"] = self.options.shared
+            tc.cache_variables["HWLOC_WITH_LIBXML2"] = self.options.with_libxml2
+            tc.generate()
+        else:
+            deps = PkgConfigDeps(self)
+            deps.generate()
+            tc = AutotoolsToolchain(self)
+            if not self.options.with_libxml2:
+                tc.configure_args.extend(["--disable-libxml2"])
+            tc.configure_args.extend(["--disable-io", "--disable-cairo"])
+            tc.generate()
+
+    def build(self):
+        if self.settings.os == "Windows":
+            cmake = CMake(self)
+            cmake.configure(build_script_folder=os.path.join("contrib", "windows-cmake"))
+            cmake.build()
+        else:
+            autotools = Autotools(self)
+            autotools.configure()
+            autotools.make()
+
+    def package(self):
+        copy(self, "COPYING", self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        if self.settings.os == "Windows":
+            cmake = CMake(self)
+            cmake.install()
+            # remove PDB files
+            rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
+        else:
+            autotools = Autotools(self)
+            autotools.install()
+            fix_apple_shared_install_name(self)
+            # remove tools
+            rmdir(self, os.path.join(self.package_folder, "bin"))
```

Comment:
in what context may these tools be used? what is the rationale behind removing them altogether?

Reply:
These tools are kind of prebuilt samples, which you can use to play with hwloc functionality.
E.g. understand topology of your system, bind threads to cores, etc. I can hardly imagine that someone needs this during build of another application. 

If a developer plays with HWLOC, it can download archive and build on his / her own, while Conan recipe is useful for integration purposes, where these tools are not needed.

Even someone needs them in future, she / he can send a PR with changes where some tools are kept back and provide arguments why it's needed. For now, since I'm not hHWLOC developer / maintainer / advanced user, I'm not the right person for these questions. What I need is to enable topology knowledge for TBB, where minimal HWLOC capabilities are required.

Reply:
Thanks a lot for the explanation, we appreciate it